harness/determined/pytorch/_callback.py

```diff
@@ -126,6 +135,16 @@ def on_training_epoch_end(self, epoch_idx: int) -> None:
         """
         pass
 
+    def on_training_workload_end(self, metrics: Dict[str, Any]) -> None:
+        """
+        Run on end of a training workload. Each workload contains a maximum of ``scheduling_unit``
+        batches, as defined in the experiment config.
```

Comment:
I recommend softening this language to make it easier for us to change it with trainer API and true epoch-based training.

"The exact length of a training workload is not guaranteed, though in the current implementation of PyTorchTrial, each workload contains a maximum ..."